### PR TITLE
Metrics as per datastore

### PR DESCRIPTION
**Pull request by mattford63**

### CODE REVIEW

File: src/kixi/heimdall/components/metrics.clj [hunk 1]
```diff
@@ -11,6 +11,61 @@
             [metrics.ring.expose :as expose]
             [taoensso.timbre :as log]))
 
+(def upper-name
+  (memoize
```
Comment (otfrom) [inline src/kixi/heimdall/components/metrics.clj]: why the memoize here and on `uri-method-stats->metric-name`?

File: src/kixi/heimdall/components/metrics.clj [hunk 2]
```diff
@@ -11,6 +11,61 @@
...
+                     safe-name
+                     (str "." status))]
+        ["resources" (safe-name uri) name]))))
+
+(defn request-ctx->metric-name
+  [ctx]
+  (uri-method-status->metric-name
+    (get-in ctx [:request :uri])
+    (:method ctx)
+    (get-in ctx [:response :status])))
```
Comment (otfrom) [inline src/kixi/heimdall/components/metrics.clj]: are we OK if any of these are nil? Do we want a different default?

Comment (mattford63) [reply src/kixi/heimdall/components/metrics.clj]: Not sure what default though:  this is lifted from datastore code that's been working happily for a good whilst.  Happy to leave as is.

File: src/kixi/heimdall/components/metrics.clj (hunk 1 above)
Comment (mattford63) [reply src/kixi/heimdall/components/metrics.clj]: not sure what you are getting at they are different functions being memoized...

Comment (otfrom) [reply src/kixi/heimdall/components/metrics.clj]: ok

**Pull request merged**